projects/frontend/data-pipelines/projects/gui/README.md

```diff
@@ -0,0 +1,126 @@
+
+You can use Frontend implementation in `projects/ui`.
+Using `npm link` you can achieve real-time development of the library without the need to upload it to repository
+
+_**Note**_: If you run directly npm install without linking it will tell you that there is no such package.
+That's why you need to link it the first time you build your development environment.
+
+Refer to [.gitlab-ci.yml](../../.gitlab-ci.yml) file for correct setup guaranteed to work by automation.
+Bug in general the steps are:
+
+0. Set npm registry to use
+   ```bash
+   $ npm config set legacy-peer-deps true
+   ```
+1. Install Angular Cli and all other dependencies except '@vdk/data-pipelines'
+   ```bash
+   npm i -g @angular/cli@13
+   ```
+   ```bash
+   npm ci --omit=optional
+   ```
+2. Build the data-pipelines UI library
+   ```bash
+   npm run build
+   ```
+3. Go to the build dir
+   ```bash
+   cd dist/data-pipelines
+   ```
+4. Link it to local repo
+   ```bash
+   npm link
+   ```
+5. Return to project dir
+   ```bash
+   cd ../../
+   ```
+6. Create symlink from build artifact to the wrapper ui (ui)
+   ```bash
+   npm link @vdk/data-pipelines
+   ```
+7. Start the wrapper ui application, serve on [http://localhost.vmware.com:4200](http://localhost.vmware.com:4200)
+   ```bash
+   npm start
+   ```
+
+## Adding boilerplate code
+
+This project is using [NX](https://nx.dev/latest/angular/getting-started/getting-started).
+It supports many plugins which add capabilities for developing different types of applications and different tools.
+These capabilities include generating applications, libraries, etc as well as the devtools to test, and build projects as well.
+
+### Generate a library
+
+Run `ng g @nrwl/angular:lib <my-lib>` to generate a library.
+
+Libraries are shareable across libraries and applications. They can be imported from `@vdk/<mylib>`.
+
+### Code scaffolding
+Run `ng generate component component-name --project ui` to generate a new component. You can also use `ng generate directive|pipe|service|class|guard|interface|enum|module --project ui`.
+> Note: Don't forget to add `--project ui` or else it will be added to the default project in your `angular.json` file.
+
+Also, this project uses [NgRx](https://ngrx.io/) for state management, you can check their [schematics](https://ngrx.io/guide/schematics) for code generation like:
+```shell
+ng generate @ngrx/schematics:effect Teams --module app.module.ts
+```
+
+### Linting
+Run `npm run lint` to execute lint via [ESLint](https://eslint.org/docs/user-guide/getting-started).
+
+### Running tests
+
+[Unit] Run `npm run test` to execute the unit tests via [Karma](https://karma-runner.github.io).
+
+_**Note**_: Code coverage report will be generated in `reports/coverage`(for the UI Wrapper and the Data Pipeline Library)
+
+_**Note**_: Code coverage report for the Data Pipeline library will be also logged in the Console(and consumed by the CI/CD)
+
+TODO re-visit when OAuth is agreed on
```

Comment:
is there a ticket for this todo?

Reply:
https://github.com/vmware/versatile-data-kit/issues/1610.
Updated comment also.